src/legacy/core_plugins/kibana/public/management/sections/index_patterns/edit_index_pattern/edit_index_pattern.js

```diff
@@ -39,132 +44,196 @@ import { I18nContext } from 'ui/i18n';
+            );
+            $scope.refreshFilters();
+            $scope.$apply();
+          }}
+        />
+      </I18nContext>,
+      node
+    );
+  });
 }
 
 function destroyScriptedFieldsTable() {
   const node = document.getElementById(REACT_SCRIPTED_FIELDS_DOM_ELEMENT_ID);
   node && unmountComponentAtNode(node);
 }
 
-function updateIndexedFieldsTable($scope, $state) {
-  if ($state.tab === 'indexedFields') {
-    $scope.$$postDigest(() => {
-      const node = document.getElementById(REACT_INDEXED_FIELDS_DOM_ELEMENT_ID);
-      if (!node) {
-        return;
-      }
-
-      render(
-        <I18nContext>
-          <IndexedFieldsTable
-            fields={$scope.fields}
-            indexPattern={$scope.indexPattern}
-            fieldFilter={$scope.fieldFilter}
-            fieldWildcardMatcher={$scope.fieldWildcardMatcher}
-            indexedFieldTypeFilter={$scope.indexedFieldTypeFilter}
-            helpers={{
-              redirectToRoute: (obj, route) => {
-                $scope.kbnUrl.redirectToRoute(obj, route);
-                $scope.$apply();
-              },
-              getFieldInfo: $scope.getFieldInfo,
-            }}
-          />
-        </I18nContext>,
-        node
-      );
-    });
-  } else {
-    destroyIndexedFieldsTable();
-  }
+function updateIndexedFieldsTable($scope) {
+  $scope.$$postDigest(() => {
+    const node = document.getElementById(REACT_INDEXED_FIELDS_DOM_ELEMENT_ID);
+    if (!node) {
+      return;
+    }
+
+    render(
+      <I18nContext>
+        <IndexedFieldsTable
+          fields={$scope.fields}
+          indexPattern={$scope.indexPattern}
+          fieldFilter={$scope.fieldFilter}
+          fieldWildcardMatcher={$scope.fieldWildcardMatcher}
+          indexedFieldTypeFilter={$scope.indexedFieldTypeFilter}
+          helpers={{
+            redirectToRoute: (obj, route) => {
+              $scope.kbnUrl.changeToRoute(obj, route);
+              $scope.$apply();
+            },
+            getFieldInfo: $scope.getFieldInfo,
+          }}
+        />
+      </I18nContext>,
+      node
+    );
+  });
 }
 
 function destroyIndexedFieldsTable() {
   const node = document.getElementById(REACT_INDEXED_FIELDS_DOM_ELEMENT_ID);
   node && unmountComponentAtNode(node);
 }
 
+function handleTabChange($scope, newTab) {
+  switch (newTab) {
+    case TAB_SCRIPTED_FIELDS:
+      destroyIndexedFieldsTable();
+      destroySourceFiltersTable();
+      return updateScriptedFieldsTable($scope);
+    case TAB_INDEXED_FIELDS:
+      destroyScriptedFieldsTable();
+      destroySourceFiltersTable();
+      return updateIndexedFieldsTable($scope);
+    case TAB_SOURCE_FILTERS:
+      destroyIndexedFieldsTable();
+      destroyScriptedFieldsTable();
+      return updateSourceFiltersTable($scope);
+  }
+}
+
+/**
+ * Create state container with sync config for tab navigation specific for edit_index_patter page
+ */
+function createEditIndexPatternPageStateContainer({ useHashedUrl }) {
```

Comment:
Maybe its worth separating it into a different file?

Reply:
Moved to different part!  and `.ts`